refgenconf/refgenconf.py

```diff
@@ -186,13 +318,71 @@ def _invert_genomes(self):
         return genomes
 
 
-def select_genome_config(filename, conf_env_vars=None):
+def _download_json(url):
+    """
+    Safely connects to the provided API endpoint and downloads the JSON formatted data
+
+    :param str url: server API endpoint
+    :return dict: served data
+    """
+    import json, requests
+    _LOGGER.debug("Downloading JSON data; querying URL: '{}'".format(url))
+    try:
+        server_resp = requests.get(url)
+    except Exception as e:
+        reason = "{}: {}".format(e.__class__.__name__, e)
+    else:
+        if server_resp.ok:
+            return json.loads(server_resp.content)
+        reason = "Response status: {}".format(server_resp.status_code)
+    _LOGGER.warning("Error querying '{}' -- {}".format(url, reason))
+
+
+def _download_url_to_file(url, filepath):
+    """
+    Download asset at given URL to given filepath.
+
+    :param str url: URL to download
+    :param str filepath: path to file to save download
+    """
+    with urllib.request.urlopen(url) as response, open(filepath, 'wb') as outf:
+        shutil.copyfileobj(response, outf)
+
+
+def _is_large_archive(size):
+    """
+    Determines if the file is large based on a string formatted as follows: 15.4GB
+
+    :param str size:  size string
+    :return bool: the decision
+    """
+    _LOGGER.debug("Checking archive size: '{}'".format(size))
+    return size.endswith("TB") or (
+            size.endswith("GB") and float("".join(c for c in size if c in '0123456789.')) > 5)
+
+
+def _list_remote(url):
+    """
+    List genomes and assets available remotely.
+
+    :param url: location or ref genome config data
+    :return str, str: text reps of remotely available genomes and assets
+    """
+    data = _read_remote_data(url)
+    rem_rgc = RefGenConf(OrderedDict({CFG_GENOMES_KEY: AttMap(data)}))
+    genomes = rem_rgc.genomes_str()
+    assets = rem_rgc.assets_str()
+    return genomes, assets
+
+
+def _read_remote_data(url):
```

Comment:
just thinking -- I think these remote_data functions should actually move up one package level into `yacman`... they will likely be useful for other yaml config stuff...

Reply:
Do you want this done before this merge/release, or after?

Reply:
up to you, if you even think it's a good idea. after is fine.